main.nf

```diff
@@ -961,6 +979,80 @@ process krona {
 ================================================================================
 */
 
+// Co-assembly: prepare grouping for MEGAHIT and pool reads for SPAdes
+if (params.coassemble_group) {
+    // short reads
+    // group and set group as new name
+    ch_short_reads_assembly
+        .groupTuple(by: 1)
+        .map { names, grp, reads ->
+                if (!params.single_end) [ "group$grp", grp, reads.collect { it[0] }, reads.collect { it[1] } ]
+                else [ "group$grp", grp, reads.collect { it[0] }, [] ] }
+        .into { ch_short_reads_megahit; ch_grouped_short_reads }
+
+    // pool short reads for SPAdes assembly
+    process pool_short_reads {
+        tag "$name"
+
+        input:
+        set val(name), val(grp), file(reads1), file(reads2) from ch_grouped_short_reads
+
+        output:
+        set val(name), val(grp), file("pooled_group${grp}*.fastq.gz") into (ch_short_reads_spadeshybrid, ch_short_reads_spades)
+
+        script:
+        if ( !params.single_end ) {
+            """
+            echo "Pooling samples: ${name}"
+            cat ${reads1} > "pooled_group${grp}_R1.fastq.gz"
+            cat ${reads2} > "pooled_group${grp}_R2.fastq.gz"
+            """
+        } else {
+            """
+            echo "Pooling samples: ${name}"
+            cat ${reads1} > "pooled_group${grp}.fastq.gz"
+            """
+        }
+    }
+
+    // long reads
+    ch_long_reads_assembly
+        .groupTuple(by: 1)
+        .map { names, grp, reads -> [ "group$grp", grp, reads.collect { it } ] }
+        .set { ch_grouped_long_reads }
+
+    // pool long reads for SPAdes assembly
+    process pool_long_reads {
+        tag "$name"
```

Comment:
as below: in case co-assembling is turned on `name` will become "group$grp", see https://github.com/skrakau/mag/blob/2c41ac0b6949819c9ed93d53fa9a2c6a1e3e8c72/main.nf#L1021

Reply:
sorry, should have mentioned it

Reply:
Its fine, just haven't seen that line